tests/foreman/api/test_rhcloud_inventory.py

```diff
@@ -334,3 +353,65 @@ def test_include_parameter_tags_setting(
             if tag['namespace'] == 'satellite_parameter':
                 assert type(tag['value']) is str
                 break
+
+
+@pytest.mark.tier3
+def test_rh_cloud_tag_values(
+    inventory_settings, organization_ak_setup, rhcloud_sat_host, rhcloud_registered_hosts
+):
+    """Verify that tag values are escaped properly when hostgroup name
+        contains " (double quote) in it.
+
+    :id: ea7cd7ca-4157-4aac-ad8e-e66b88740ce3
+
+    :Steps:
+        1. Create Hostcollection with name containing double quotes.
+        2. Register a content host with satellite.
+        3. Add a content host to hostgroup.
+        4. Generate inventory report.
+        5. Assert that generated report contains valid json file.
+        6. Assert that hostcollection tag value is escaped properly.
+
+    :expectedresults:
+        1. Valid json report is created.
+        2. Tag value is escaped properly.
+
+    :BZ: 1874587, 1874619
+
+    :CaseAutomation: Automated
+    """
+    org, ak = organization_ak_setup
+
+    host_col_name = gen_string('alpha')
+    host_name = rhcloud_registered_hosts[0].hostname
+    host = rhcloud_sat_host.api.Host().search(query={'search': host_name})[0]
+    host_collection = rhcloud_sat_host.api.HostCollection(
+        organization=org, name=f'"{host_col_name}"', host=[host]
+    ).create()
+
+    assert len(host_collection.host) == 1
+    local_report_path = robottelo_tmp_dir.joinpath(f'{gen_alphanumeric()}_{org.id}.tar.xz')
+    # Generate report
+    timestamp = datetime.utcnow().strftime('%Y-%m-%d %H:%M')
+    rhcloud_sat_host.api.Organization(id=org.id).rh_cloud_generate_report()
+    wait_for(
+        lambda: rhcloud_sat_host.api.ForemanTask()
+        .search(query={'search': f'{generate_report_task} and started_at >= "{timestamp}"'})[0]
+        .result
+        == 'success',
+        timeout=400,
+        delay=15,
+        silent_failure=True,
+        handle_exception=True,
+    )
```

Comment:
you can use `generate_inventory_report(rhcloud_sat_host, org)` here.

Reply:
Got it!